File: hello-events/src/main/java/com/hello/Entity/Event.java
package com.hello.Entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Date;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor

public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private LocalDateTime date;
    private String location;
    private String category;
    private int availableTickets;

    @OneToMany(cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Reservation> reservations;


}

File: events-frontend/src/app/core/models/event.ts
import {EventCategory} from "../enums/event-category.enum";
import {Reservation} from "./reservation";

export class Event {
  id!: number;
  name!: string;
  description!: string;
  date!: Date;
  location!: string;
  category!: EventCategory;
  availableTickets!: number;
  reservations!: Reservation[];
}
